File: test/file/scavenge/basic_test.pl
#!/usr/local/bin/perl

# @OSF_COPYRIGHT@
# COPYRIGHT NOTICE
# Copyright (c) 1990, 1991, 1992, 1993 Open Software Foundation, Inc.
# ALL RIGHTS RESERVED (DCE).  See the file named COPYRIGHT.DCE in the
# src directory for the full copyright text.
#
# HISTORY
# $Log: basic_test.pl,v $
# Revision 1.1.8.1  1996/10/17  18:35:20  damon
# 	Submitting on behalf of Transarc
# 	[1996/10/17  16:57:32  damon]
#
# Revision 1.1.3.1  1994/07/13  22:32:30  devsrc
# 	merged with bl-10
# 	[1994/06/29  12:15:54  devsrc]
# 
# 	moving from file/episode to test/file
# 	[1994/03/22  21:02:40  delgado]
# 
# 	Embedding copyright notices
# 	[1993/01/19  14:13:34  zeliff]
# 
# 	Transarc delta: jdp-ot5076-lfs-add-basic_test.pl 1.1
# 	  Selected comments:
# 	    Add basic_test.pl--a place for putting general scavenger tests.
# 	    This file is a place for putting general scavenger tests.  Its first test
# 	    bashes the bOffset field of a COW anode.
# 	[1992/09/23  17:50:03  jaffe]
# 
# $EndLog$
#

# Copyright (C) 1992 Transarc Corporation - All rights reserved

require 'efs_query';


#
# Find a COW anode and bash its 'bOffset' field
#
if (&FindObject(*GetAggr, 1, *GetAVLAnode, 1, *GetFileSet, 1,
		*GetFSAnode,  q/$anode{'bOffset'} != 0/)) {
    &StartTest();
    &SetAnode('bOffset', 515);
    &EndTest();
}


&Cleanup();

File: test/file/recovery/basictest.rs
# @OSF_COPYRIGHT@
# COPYRIGHT NOTICE
# Copyright (c) 1990, 1991, 1992, 1993 Open Software Foundation, Inc.
# ALL RIGHTS RESERVED (DCE).  See the file named COPYRIGHT.DCE for
# the full copyright text.
#
# HISTORY
# $Log: basictest.rs,v $
# Revision 1.1.8.1  1996/10/17  18:33:55  damon
# 	Submitting on behalf of Transarc
# 	[1996/10/17  16:56:50  damon]
#
# Revision 1.1.3.1  1994/07/13  22:31:40  devsrc
# 	moving files from src/file/episode to test/file
# 	[1994/03/22  20:40:17  delgado]
# 
# 	    delta bob-add-rcx 1.1
# 	    [1992/05/04  19:33:50  jdp]
# 	[1992/05/05  18:27:49  jdp]
# 
# Revision 1.1.8.1  1994/02/04  20:14:31  devsrc
# 	Merged from 1.0.3a to 1.1
# 	[1994/02/04  15:10:58  devsrc]
# 
# Revision 1.1.6.1  1994/01/28  20:43:40  annie
# 	expand OSF copyright
# 	[1994/01/28  20:42:15  annie]
# 
# 	expand OSF copyright
# 
# Revision 1.1.4.2  1993/04/14  21:16:32  jaffe
# 	Sync with Transarc Source
# 	[1993/04/14  21:03:39  jaffe]
# 
# $EndLog$

createfiles 300 abcdefg 
mkdirs 20 foo 
mvpaths 20 foo bar 
mkdir zoo
mvpaths 20 bar zoo/bar 
mvpaths 20 zoo/bar this-is-a-long-entry-name 
mvpaths 20 this-is-a-long-entry-name short 
mvpaths 20 short x 
mvpaths 20 x longer 
mkdir soup
mvpaths 20 longer soup/longer 
mvpaths 20 soup/longer zoo/longer 
mvpaths 20 zoo/longer soup/a 
